src/App/App.js

```diff
@@ -29,10 +31,18 @@ class App extends Component {
   //   this.setState({ isLoginView: !this.state.isLoginView})
   // }
 
+  findMovieIndexById(id) {
+    let foundMovie = this.state.movies.find(movie => {
+      return movie.id === id;
+    })
+    return this.state.movies.indexOf(foundMovie)
+  }
+
   render() { 
```

Comment:
Consider which parts of this you want to control through conditional rendering, and what you want to control through Router :) And don't forget to delete dead code!